core/domain/user_services.py

```diff
@@ -713,8 +716,27 @@ def get_usernames(user_ids):
         If a user_id does not exist, the corresponding entry in the
         returned list is None.
     """
-    users_settings = get_users_settings(user_ids)
-    return [us.username if us else None for us in users_settings]
+    usernames = [''] * len(user_ids)
+    non_system_user_indices = []
+    non_system_user_ids = []
+    for index, user_id in enumerate(user_ids):
+        if user_id in SYSTEM_USERS:
+            usernames[index] = SYSTEM_USERS[user_id]
+        else:
+            non_system_user_indices.append(index)
+            non_system_user_ids.append(user_id)
+
+    user_settings = get_users_settings(non_system_user_ids)
+
+    user_settings_index = 0
+    for index in non_system_user_indices:
```

Comment:
Not sure you need to keep track of two indices. How about this?
```
for index, user_settings in enumerate(non_system_users_settings):
    if user_settings:
        usernames[non_system_user_indices] = user_settings.username
```

Reply:
Yup thanks :)